test/gtest/uct/ib/test_rc.h

```diff
@@ -141,6 +144,62 @@ class test_rc_flow_control : public test_rc {
 };
 
 
+#if IBV_HW_TM
+class test_rc_mp_xrq : public uct_test {
+public:
+    static const uint64_t SEND_SEED = 0xa1a1a1a1a1a1a1a1ul;
+    static const uint64_t AM_ID     = 1;
+    typedef void (test_rc_mp_xrq::*send_func)(mapped_buffer*);
+
+    virtual void init();
+
+    test_rc_mp_xrq();
+
+    uct_rc_mlx5_iface_common_t* rc_mlx5_iface(entity &e) {
+        return ucs_derived_of(e.iface(), uct_rc_mlx5_iface_common_t);
+    }
+
+    void send_eager_bcopy(mapped_buffer *buf);
+    void send_eager_zcopy(mapped_buffer *buf);
+    void send_rndv_zcopy(mapped_buffer *buf);
+    void send_rndv_request(mapped_buffer *buf);
+    void test_common(send_func sfunc, size_t num_segs, size_t exp_segs = 1,
+                     bool exp_val = true);
+
+    static ucs_status_t am_handler(void *arg, void *data, size_t length,
+                                   unsigned flags);
+
+    static ucs_status_t unexp_eager(void *arg, void *data, size_t length,
+                                    unsigned flags, uct_tag_t stag,
+                                    uint64_t imm, void **context);
+
+    static ucs_status_t unexp_rndv(void *arg, unsigned flags, uint64_t stag,
+                                   const void *header, unsigned header_length,
+                                   uint64_t remote_addr, size_t length,
+                                   const void *rkey_buf);
+
+
+protected:
+    bool             m_first_received;
+    bool             m_last_received;
+    uct_completion_t m_uct_comp;
```

Comment:
`m_first_received`, `m_last_received`, `m_uct_comp` can be declared as `private` fields